packages/fulfillment/src/services/fulfillment-module-service.ts

```diff
@@ -504,6 +588,55 @@ export default class FulfillmentModuleService<
       : createdShippingOptionRules[0]
   }
 
+  @InjectManager("baseRepository_")
+  async createFulfillment(
+    data: FulfillmentTypes.CreateFulfillmentDTO,
+    @MedusaContext() sharedContext: Context = {}
+  ): Promise<FulfillmentTypes.FulfillmentDTO> {
+    const { order, ...fulfillmentDataToCreate } = data
+
+    const fulfillment = await this.fulfillmentService_.create(
+      fulfillmentDataToCreate,
+      sharedContext
+    )
+
+    const {
+      items,
+      data: fulfillmentData,
+      provider_id,
+      ...fulfillmentRest
+    } = fulfillment
+
+    let fulfillmentThirdPartyData!: any
+    try {
+      fulfillmentThirdPartyData =
+        await this.serviceProviderService_.createFulfillment(
+          provider_id,
+          fulfillmentData || {},
+          items.map((i) => i),
+          order,
+          fulfillmentRest
```

Comment:
thought: See other comment. Perhaps, we should rethink the shape of objects passed to the provider here. For example, the properties we need from `order` could be extracted, so it is decoupled from the actual order concept.

Reply:
yes I think we need to rethink the entire API, I ve a todo somewhere in the abstract of the interface to rework all dtos and stuff